tests/tensorflow/helpers.py

```diff
@@ -125,18 +129,18 @@ def get_coco_dataset_builders(config, num_devices, **kwargs):
                                                is_train=True,
                                                num_devices=num_devices))
 
-        if kwargs.get('calibration', False):
-            config_ = config.deepcopy()
-            config_.batch_size = builders[0].batch_size
-            builders.append(MockCOCODatasetBuilder(config=config_,
-                                                   is_train=True,
-                                                   num_devices=1))
-
     if kwargs.get('validation', False):
         builders.append(MockCOCODatasetBuilder(config=config,
                                                is_train=False,
                                                num_devices=num_devices))
 
+    if kwargs.get('calibration', False):
+        config_ = config.deepcopy()
+        config_.batch_size = builders[0].batch_size
+        builders.append(MockCOCODatasetBuilder(config=config_,
+                                               is_train=True,
+                                               num_devices=1))
```

Comment:
calibration dataset is created not for train mode only, if calibration=True is specified (segmentation sample case)